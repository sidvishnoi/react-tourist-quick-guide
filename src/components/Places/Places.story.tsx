import { object, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import * as React from 'react';
import Places, { PlacesProps } from '.';

const stories = storiesOf('Places', module);
stories.addDecorator(withKnobs);

stories.add('places', () => {
  const props: PlacesProps = {
    places: object('items', [
      { title: 'Place one', link: '#', summary: 'this is a summary' },
      { title: 'Place two', link: '#', summary: 'this is a longer summary' },
      { title: 'Place third', link: '#', summary: 'this is a summary' },
      {
        link: '#',
        summary: 'this is a summary',
        title: 'Place fourth is long',
      },
      { title: 'Place five', link: '#', summary: 'this is a summary' },
    ]),
  };

  return <Places places={props.places} />;
});
