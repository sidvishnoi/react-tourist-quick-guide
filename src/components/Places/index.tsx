import * as React from 'react';

import Link from '../Link';
import Text from '../Text';
import UnorderedList from '../UnorderedList';

export interface PlacesProps {
  places: {
    title: string;
    link: string;
    summary: string;
  }[];
  [prop: string]: any;
}

export default function(props: PlacesProps) {
  const { places } = props;
  return (
    <div>
      <Text as="h3" size="fsize_05">
        Top tourist places:
      </Text>
      <UnorderedList>
        {places.map((place, i) => (
          <span key={i}>
            <Link href={place.link}>{place.title}</Link>:{' '}
            <em>{place.summary}</em>
          </span>
        ))}
      </UnorderedList>
    </div>
  );
}
