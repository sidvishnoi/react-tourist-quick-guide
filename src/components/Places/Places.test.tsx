import { mount } from 'enzyme';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

import Places from '.';

describe('Components - List', () => {
  const places = [{ name: 'one', link: '#1' }, { name: 'two', link: '#2' }];

  it('matches snapshot', () => {
    const tree = renderer.create(<Places places={places} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders title', () => {
    const output = mount(<Places places={places} />);
    expect(output.find('h3').text()).toEqual('Top 2 tourist places:');
  });

  it('should display list of places', () => {
    const output = mount(<Places places={places} />);
    expect(output.find('li').map(p => p.text())).toEqual(
      places.map(place => place.name),
    );
    expect(output.find('li a').map(p => p.prop('href'))).toEqual(
      places.map(place => place.link),
    );
  });
});
