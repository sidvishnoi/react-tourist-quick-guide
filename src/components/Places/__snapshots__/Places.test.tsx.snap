// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Places matches snapshot 1`] = `
.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #333;
}

.c3:hover,
.c3:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  font-family: serif;
  color: #000;
  font-size: 1.25rem;
  line-height: 1.625rem;
  margin: 0.2rem 0;
}

.c2 {
  margin: 0.2rem 0;
}

.c1 {
  margin: 0.5rem 0 0 0.5rem;
}

@media print {
  .c3 {
    color: #000;
  }
}

<div>
  <h3
    className="c0"
    color="#000"
    size="fsize_05"
  >
    Top tourist places:
  </h3>
  <ul
    className="c1"
  >
    <li
      className="c2"
    >
      <span>
        <a
          className="c3"
          href="#1"
        >
          one
        </a>
        :
         
        <em>
          foo
        </em>
      </span>
    </li>
    <li
      className="c2"
    >
      <span>
        <a
          className="c3"
          href="#2"
        >
          two
        </a>
        :
         
        <em>
          bar
        </em>
      </span>
    </li>
  </ul>
</div>
`;
